### PR TITLE
utils: fix build and add a method to target specific props and extend

### PR DESCRIPTION
**Pull request by teleaziz**

### CODE REVIEW

File: packages/utils/README.md [hunk 1]
```diff
@@ -50,4 +50,100 @@ export async function getStaticProps(context) {
...
+```
+This component will update to fetch product data browser side by query chosen in the Builder editor. But if you want this to render server side you need `props.products` to be passed with the needed data on render.
+
+To accomplish this, you can use `extendAsyncProps`. 
+
+Here is a usage example with Next.js:
+
+```tsx
+import { Builder, builder } from '@builder.io/react';
+import { extendAsyncProps } from '@builder.io/utils';
```
Comment (mandx) [inline packages/utils/README.md:130]: In this example, this is imported, but not used?

File: packages/utils/README.md [hunk 2]
```diff
@@ -50,4 +50,100 @@ export async function getStaticProps(context) {
...
+
+export default function MyPage(props) {
+  return <BuilderComponent model="page" content={props.content} />
+}
+
+export async function getStaticProps(context) {
+  const content = await builder.get('page', { url: context.resolvedUrl }).promise();
+  
+  await getAsyncProps(content, {
```
Comment (mandx) [inline packages/utils/README.md:139]:
```suggestion
  await extendAsyncProps(content, {
```
Is this what you mean?

File: packages/utils/README.md [hunk 3]
```diff
@@ -50,4 +50,100 @@ export async function getStaticProps(context) {
 
   return { props: { content } }
 }
+```
+
+## extendAsyncProps
+
+`extendAsyncProps` is similar to `getAsyncProps` except it look up the inputs by name across components and map it's props to the result of the asynchronus mapper function passed.
```
Comment (mandx) [inline packages/utils/README.md:57]:
```suggestion
`extendAsyncProps` is similar to `getAsyncProps` except it looks up the inputs by name across components and map its props to the result of the asynchronus mapper function passed.
```
Also, maybe reword this in simpler terms?

> map its props to the result of the asynchronus mapper function passed.

It might takes a few re-reads to understand properly...

File: packages/utils/README.md [hunk 4]
```diff
@@ -50,4 +50,100 @@ export async function getStaticProps(context) {
...
\ No newline at end of file
+```
+
+## extendAsyncProps
+
+`extendAsyncProps` is similar to `getAsyncProps` except it looks up the inputs by name across components and map its props to the result of the asynchronus mapper function passed.
+
+### Example
+
+Say you have multiple components that takes a query input and you want to resolve this query asynchronusly before rendering on the server
```
Comment (mrkoreye) [inline packages/utils/README.md:61]:
```suggestion
Say you have multiple components that take a query input and you want to resolve this query asynchronusly before rendering on the server
```

**Review by mrkoreye — APPROVED** [review]
